src/libs/antares/inifile/inifile.cpp

```diff
@@ -253,95 +147,143 @@ uint IniFile::Section::size() const
     return count;
 }
 
-bool IniFile::open(const AnyString& filename, bool warnings)
+static bool isStartingSection(std::string line)
 {
-    // clear
-    clear();
-    pFilename = filename;
+    boost::trim(line);
+    return line.starts_with("[") && line.ends_with("]");
+}
+
+static std::string getSectionName(const std::string& line)
+{
+    std::vector<std::string> splitLine;
+    boost::split(splitLine, line, boost::is_any_of("[]"));
+    return splitLine[1];
+}
+
+static bool isProperty(const std::string& line)
+{
+    return std::ranges::count(line.begin(), line.end(), '=') == 1;
+}
 
-    // Load the file
-    IO::File::Stream file;
-    if (file.open(filename))
+static IniFile::Property getProperty(std::string line)
+{
+    boost::trim(line);
+    std::vector<std::string> splitLine;
+    boost::split(splitLine, line, boost::is_any_of("="));
+
+    return IniFile::Property(splitLine[0], splitLine[1]);
+}
+
+static bool isComment(std::string line)
+{
+    boost::trim_left(line);
+    return line.starts_with("#") || line.starts_with(";");
+}
+
+bool isEmpty(std::string line)
+{
+    boost::trim(line);
+    return line.empty();
+}
+
+bool IniFile::readStream(std::istream& in_stream)
+{
+    std::string line;
+    IniFile::Section* currentSection(nullptr);
+    uint64_t read = 0;
+    while (std::getline(in_stream, line))
     {
-        enum
+        read += line.size();
+        if (isStartingSection(line))
         {
-            lineHardLimit = 2048
-        };
-
-        auto* line = new char[lineHardLimit];
+            currentSection = addSection(getSectionName(line));
+            continue;
+        }
 
-        IniFile::Section* lastSection = nullptr;
-        uint64_t read = 0;
+        if (isProperty(line))
+        {
+            // Note : if a property not in a section, it's simply skipped
+            if (currentSection)
+            {
+                IniFile::Property p = getProperty(line);
+                currentSection->add(p.key, p.value);
```

Comment:
```suggestion
                currentSection->add(Property(line));
```

Reply:
If we do that, we couple reading concern (a line) and business logic (Property).
As it's done now, **getProperty** is in the reading area and depend on type **Property** (business).
If we introduce **Property(std::string line)** reading puts a step into business logic (dependency in the other direction).

Reply:
If we want to merge those 2 lines in one, we could have a new Section::add function, taking a Property as argument, so we could have : 
`currentSection->add(getProperty(line));`